### PR TITLE
Update to k8s 1.19

### PR DESCRIPTION
**Pull request by erikwilson**
Proposed changes
======

k8s update to v1.19.0

Types of changes
======

Component Upgrade

Verification
======

Needs certified conformance test for cncf

Linked Issues
======
 
For #2020

Further comments
======

### CODE REVIEW

**Comment by brandond** [comment]
Are we going to follow k/k up to golang 1.15?

**Comment by erikwilson** [comment]
The minimum requirement for k8s 1.19 is golang 1.14.4, and since 1.15 is not officially out yet I think we should stick with 1.14.

File: go.mod [hunk 1]
```diff
@@ -27,73 +27,64 @@ replace (
...
+	k8s.io/component-base => github.com/erikwilson/kubernetes/staging/src/k8s.io/component-base v1.19.0-rc3-k3s1
+	k8s.io/cri-api => github.com/erikwilson/kubernetes/staging/src/k8s.io/cri-api v1.19.0-rc3-k3s1
+	k8s.io/csi-translation-lib => github.com/erikwilson/kubernetes/staging/src/k8s.io/csi-translation-lib v1.19.0-rc3-k3s1
+	k8s.io/kube-aggregator => github.com/erikwilson/kubernetes/staging/src/k8s.io/kube-aggregator v1.19.0-rc3-k3s1
+	k8s.io/kube-controller-manager => github.com/erikwilson/kubernetes/staging/src/k8s.io/kube-controller-manager v1.19.0-rc3-k3s1
+	k8s.io/kube-proxy => github.com/erikwilson/kubernetes/staging/src/k8s.io/kube-proxy v1.19.0-rc3-k3s1
+	k8s.io/kube-scheduler => github.com/erikwilson/kubernetes/staging/src/k8s.io/kube-scheduler v1.19.0-rc3-k3s1
+	k8s.io/kubectl => github.com/erikwilson/kubernetes/staging/src/k8s.io/kubectl v1.19.0-rc3-k3s1
+	k8s.io/kubelet => github.com/erikwilson/kubernetes/staging/src/k8s.io/kubelet v1.19.0-rc3-k3s1
+	k8s.io/kubernetes => github.com/erikwilson/kubernetes v1.19.0-rc3-k3s1
```
Comment (AkihiroSuda) [inline go.mod:53]: Could you update my patches with https://github.com/rootless-containers/usernetes/tree/master/src/patches/kubernetes ?

Not a blocker for 1.19, but will be needed to support rootless cgroup2

Comment (erikwilson) [reply go.mod:53]: Absolutely, thanks for pointing this out! :)

**Comment by brandond** [comment]
If you're waiting for a LGTM to push tags to rancher/kubernetes, then: **LGTM**

File: Dockerfile.manifest [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.13.15-alpine3.12
+FROM golang:1.14.6-alpine3.12
```
Comment (brandond) [inline Dockerfile.manifest:1]: Upstream is building with either 1.13 or 1.15; do we feel safe using 1.14?

Comment (brandond) [reply Dockerfile.manifest:1]: https://github.com/kubernetes/kubernetes/commit/aa2bb18a2b1c8d2f42bda82f0377f6415dbc6944

**Comment by davidnuzik** [comment]
@warmchang go was bumpted to 1.15 you can see the commit here https://github.com/rancher/k3s/pull/2098/commits/debf8005ca0af187f4437fb003980768493e1ff1 or look above

**Comment by warmchang** [comment]
@davidnuzik Actually I want to say whether we need to change the go.mod from "go 1.13" to "go 1.15":
https://github.com/rancher/k3s/blob/447097a597532ae710d22ff471c97d845720e1cf/go.mod#L3

**Comment by brandond** [comment]
We probably could bump that as well, but it shouldn't be required as per https://github.com/golang/go/issues/30791#issuecomment-472217112.

**Pull request merged**